arviz/wrappers/wrap_pystan.py

```diff
@@ -55,3 +55,85 @@ def get_inference_data(self, fit):
     def log_likelihood__i(self, excluded_obs_log_like, idata__i):
         """Retrieve the log likelihood of the excluded observations from ``idata__i``."""
         return idata__i.log_likelihood[excluded_obs_log_like]
+
+
+class PyStan3SamplingWrapper(SamplingWrapper):
+    """PyStan3 sampling wrapper base class.
+
+    See the documentation on  :class:`~arviz.SamplingWrapper` for a more detailed
+    description. An example of ``PyStan3SamplingWrapper`` usage can be found
+    in the :ref:`pystan3_refitting` notebook.
+
+    Warnings
+    --------
+    Sampling wrappers are an experimental feature in a very early stage. Please use them
+    with caution.
+    """
+
+    def __init__(
```

Comment:
Not sure renaming model to model code is worth redefining init, in pymc3 for example the "model" is actually a function that returns the model object to then do the `with model:...`. Maybe we can change the name in the base sampler to something more generic?